lib/core/src/sdk.rs

```diff
@@ -1701,10 +1707,16 @@ impl LiquidSdk {
         let receiver_amount_sat = user_lockup_amount_sat - fees_sat;
         let claim_address = self.onchain_wallet.next_unused_address().await?.to_string();
 
+        // Reserve this address until the timeout block height
+        self.persister.insert_or_update_reserved_address(
```

Comment:
@dangeross why do we need to cache claim_address in the pool? Why do we care if such claim address is reused?

Reply:
If the swap is never initiated (btc sent to lockup address) and the swap expires, then this claim address can be reused elsewhere.

Reply:
But we only care if it is used as part of magic hints and for that we already reserve addresses. I think we don't care if it is used in other places (other claims, refunds, etc...) don't we?
Also we can defer the claim creation to the actual point we know we need to claim (btc sent to lockup address) but even if not, as stated above, we should be fine.

Reply:
Makes sense, now deferring getting the incoming claim address [990ff5c](https://github.com/breez/breez-sdk-liquid/pull/531/commits/990ff5ca99052fcc50aa300018c438d906080152)